Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -0,0 +1,351 @@
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        reloadTopSites()
+
+        reloadRecentHistory()
+
+        view.addSubview(tableView)
+        tableView.snp_makeConstraints { (make) in
+            make.edges.equalTo(self.view)
+        }
+    }
+
+    override func traitCollectionDidChange(previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        self.topSiteHandler.currentTraits = self.traitCollection
+    }
+
+}
+
+// MARK: -  Section management
+extension ActivityStreamPanel {
+
+    enum Section: Int {
+        case TopSites
+        case History
+
+        static let count = 2
+
+        var title: String? {
+            switch self {
+            case .History: return "Recent Activity"
+            case .TopSites: return nil
+            }
+        }
+
+        var headerHeight: CGFloat {
+            switch self {
+            case .History: return 40
+            case .TopSites: return 0
+            }
+        }
+
+        func cellHeight(traits: UITraitCollection) -> CGFloat {
+            switch self {
+            case .History: return UITableViewAutomaticDimension
+            case .TopSites:
+                if traits.horizontalSizeClass == .Compact && traits.verticalSizeClass == .Regular {
+                    return ASPanelUX.TopSiteDoubleRowHeight
+                } else {
+                    return ASPanelUX.TopSiteSingleRowHeight
+                }
+            }
+        }
+
+        var headerView: UIView? {
+            switch self {
+            case .History:
+                let view = ASHeaderView()
+                view.title = "Recent Activity"
+                return view
+            case .TopSites:
+                return nil
+            }
+        }
+
+        var cellIdentifier: String {
+            switch self {
+            case .TopSites: return "TopSite"
+            case .History: return "Cell"
+            }
+        }
+
+        init(at indexPath: NSIndexPath) {
+            self.init(rawValue: indexPath.section)!
+        }
+
+        init(_ section: Int) {
+            self.init(rawValue: section)!
+        }
+    }
+
+}
+
+// MARK: -  Tableview Delegate
+extension ActivityStreamPanel: UITableViewDelegate {
+
+    func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return Section(section).headerHeight
+    }
+
+    func tableView(tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        return Section(section).headerView
+    }
+
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return Section(indexPath.section).cellHeight(self.traitCollection)
+    }
+
+    func showSiteWithURL(url: NSURL) {
```

Comment:
Nit: I would make this private to help identify it has a 'helper' method instead of something that is from the delegate.